src/App.tsx

```diff
@@ -1,19 +1,32 @@
+import { Suspense, lazy } from 'react';
 import { BrowserRouter, Route, Routes } from 'react-router-dom';
 import Main from './pages/Main/Main';
 import Smartphones from './pages/Smartphones/Smartphones';
 import Laptops from './pages/Laptops/Laptops';
 import Viewed from './pages/Viewed/Viewed';
+const SingleProduct = lazy(() => import('./pages/SingleProduct/index'));
```

Comment:
It's better do define variable after the import statement. Also It will be a good idea to import lazy component from SingleProduct. 
Something like SingleProductLazy